pkg/proxy/util/utils_test.go

```diff
@@ -28,6 +29,44 @@ import (
 	fake "k8s.io/kubernetes/pkg/proxy/util/testing"
 )
 
+func TestValidateWorks(t *testing.T) {
+	if isValidEndpoint("", 0) {
+		t.Errorf("Didn't fail for empty set")
+	}
+	if isValidEndpoint("foobar", 0) {
+		t.Errorf("Didn't fail with invalid port")
+	}
+	if isValidEndpoint("foobar", -1) {
+		t.Errorf("Didn't fail with a negative port")
+	}
+	if !isValidEndpoint("foobar", 8080) {
+		t.Errorf("Failed a valid config.")
+	}
+}
+
+func TestBuildPortsToEndpointsMap(t *testing.T) {
+	endpoints := &v1.Endpoints{
+		ObjectMeta: metav1.ObjectMeta{Name: "foo", Namespace: "testnamespace"},
+		Subsets: []v1.EndpointSubset{{
+			Addresses: []v1.EndpointAddress{{IP: "endpoint1"}},
+			Ports:     []v1.EndpointPort{{Name: "p", Port: 40}},
+		},
+			{
```

Comment:
hm... still weird indentation... oh, it's because you did "`EndpointSubset{{` above rather than putting a newline between the `{`s.

Also, I'd still like a slightly more complete test case. Can you grab https://github.com/kubernetes/kubernetes/commit/819ec958.patch, squash it into your commit, and then re-push?

Reply:
@danwinship 
I have pushed with suggested change.